typescript/infra/scripts/deploy-agents.ts

```diff
@@ -6,16 +6,19 @@ async function deploy() {
   const environment = await getEnvironment();
   const agentConfig = await getAgentConfig(environment);
   const domainNames = await getDomainNames(environment);
-  await Promise.all(
-    domainNames.map((name) => {
-      return runAgentHelmCommand(
-        HelmCommand.InstallOrUpgrade,
-        agentConfig,
-        name,
-        domainNames,
-      );
-    }),
-  );
+
+  // We intentionally do not Promise.all here to avoid race conditions that could result
+  // in attempting to create a user or key multiple times. This was found to happen in
+  // situations where agents for different domains would share the same AWS user or
+  // AWS KMS key.
+  for (const name of domainNames) {
```

Comment:
What if we just created a simple script that created all the keys ahead of time? I.e. modify the `create-keys` script?

Reply:
Agreed this is the best thing to do, filed this https://github.com/abacus-network/abacus-monorepo/issues/363